comp/forwarder/defaultforwarder/domain_forwarder.go

```diff
@@ -290,6 +291,24 @@ func (f *domainForwarder) sendHTTPTransactions(t transaction.Transaction) {
 		}
 	}
 
+	// TODO: why is isMRF not set when we expect it? this is a hack for local testing only
+	isMrf := strings.Contains(f.domain, ":8083")
```

Comment:
I did not dig for long so sorry if completely out of scope, but could it be because of the configured site? https://github.com/DataDog/datadog-agent/blob/30ff86bdef40dcb1699b38c5fec7e95f2145a548/comp/forwarder/defaultforwarder/default_forwarder.go#L278-L281
When testing locally, the site/domain generation may be leading to issues here since the domain is not "generated" on local domains etc. AFAIR

Reply:
Good thought, this seems correct to me. The forwarder is not marked as MRF because I am setting `dd_url` instead of `site` when testing locally. This seems like something to fix, what do you think @tobz?

Reply:
It does seem like we should/need to fix this, yeah.

My naive thought is that we ought to just be using `GetMRFInfraEndpoint` instead of building the URL manually with `BuildURLWithPrefix`, since that should match up with how we populate `options.DomainResolvers` in the first place: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/utils/endpoints.go#L103-L109